packages/primitives/heading/src/Root.native.tsx

```diff
@@ -31,50 +42,96 @@ export const Heading = component(
     shouldPreventWrap,
     shouldHideOverflow,
   }) => {
-    const style: TStyle = {
+    const textStyle: TStyle = {
       backgroundColor: 'transparent',
       lineHeight,
       fontFamily,
       fontWeight,
       fontSize,
       letterSpacing,
+      _nativeOnly: {},
+    }
+
+    const viewStyle: TStyle = {
+      paddingBottom: blockEnd,
+      paddingLeft: inlineStart,
+      paddingRight: inlineEnd,
+      paddingTop: blockStart,
     }
 
     if (isColor(color)) {
-      style.color = colorToString(color)
+      textStyle.color = colorToString(color)
+    }
+
+    if (align) {
```

Comment:
why not elegir?

Reply:
We don't seem to be using `elegir` in primitives. I don't know why exactly, but I do suspect it has to do with being minimalistic and fast, since primitives will be used everywhere. Notice how we are also mutating the objects, something we don't do anywhere else.